lib/Sema/TypeCheckDecl.cpp

```diff
@@ -696,6 +696,165 @@ ExistentialTypeSupportedRequest::evaluate(Evaluator &evaluator,
   return true;
 }
 
+static bool hasThrowingFunctionClosureParameter(CanType type) {
```

Comment:
I think this belongs in TypeCheckEffects.cpp along with the rest of the rethrows-checking